plugins/statuslines/lualine.nix

```
@@ -1,260 +1,281 @@
+          ];
+          isOldType = def: lib.any (name: lib.elem name oldAttrs) (lib.attrNames def);
+          oldType = addCheck (attrsOf anything) isOldType // {
+            description = "attribute set containing ${lib.concatStringsSep ", " oldAttrs}";
+          };
+          coerceFn =
+            option:
+            if option != null && lib.isAttrs option then
+              let
+                # NOTE: Handle deprecated name property
+                # This was used to create an unkeyed attribute for a component name
+                optionWithNameTransformed =
+                  if lib.hasAttr "name" option then
+                    lib.removeAttrs (option // { "__unkeyed" = option.name; }) [ "name" ]
+                  else
+                    option;
+
+                # NOTE: Handle deprecated nested icon option
+                # This was used to create an unkeyed attribute for an icon with options
+                optionWithIconTransformed =
+                  if
+                    lib.hasAttr "icon" optionWithNameTransformed
+                    && lib.isAttrs optionWithNameTransformed.icon
+                    && lib.hasAttr "icon" optionWithNameTransformed.icon
+                  then
+                    lib.removeAttrs (
+                      optionWithNameTransformed.icon // { "__unkeyed" = optionWithNameTransformed.icon.icon; }
+                    ) [ "icon" ]
+                  else
+                    optionWithNameTransformed;
+
+                # NOTE: Handles the deprecated extraConfig option
+                # Old freeform option was just merged into the existing config
+                flattenedExtraConfig = optionWithIconTransformed.extraConfig or { };
+                optionWithoutExtraConfig = lib.removeAttrs optionWithIconTransformed [ "extraConfig" ];
+                finalTransformedValue = optionWithoutExtraConfig // flattenedExtraConfig;
+              in
+              finalTransformedValue
+            else
+              option;
+          newType = submodule {
+            freeformType = either str (attrsOf anything);
+
+            options = {
+              icons_enabled = defaultNullOpts.mkBool true ''
+                Enables the display of icons alongside the component.
+              '';
+
+              icon = lib.nixvim.mkNullOrOption' {
+                type = either str (attrsOf anything);
+                description = "Defines the icon to be displayed in front of the component.";
+              };
+
+              separator = mkSeparatorsOption { };
+
+              color = defaultNullOpts.mkNullable (either attrs str) null ''
+                Defines a custom color for the component.
+
+                Default is the color defined by your theme for that section and mode.
+              '';
+
+              padding = defaultNullOpts.mkNullable (either int (submodule {
+                # In case they add support for top/bottom padding
                 freeformType = attrsOf anything;
```

I wouldn't normally suggest this, but I think here we could be confident of:
```suggestion
                freeformType = attrsOf int;
```